pkg/destroy/gcp/gcp.go

```diff
@@ -151,3 +160,43 @@ func isNoOp(err error) bool {
 func (o *ClusterUninstaller) contextWithTimeout() (context.Context, context.CancelFunc) {
 	return context.WithTimeout(o.Context, defaultTimeout)
 }
+
+// requestID returns a UID for a given item identifier. Unless the ID is reset, the
+// same requestID will be returned every time for a given item.
+func (o *ClusterUninstaller) requestID(identifier ...string) string {
```

Comment:
We shouldn't be add this caching functions directly onto `ClusterUninstaller`

a separate type that does the caching and keeps tracks.. etc, is not testable and easy to reason.
we can keep these functions on ClusterUninstaller but these should be internal handing off..

Reply:
makes sense, will update